src/client/game.cpp

```diff
@@ -1466,11 +1466,18 @@ bool Game::connectToServer(const GameStartData &start_data,
 		return false;
 	}
 
-	client = new Client(start_data.name.c_str(),
-			start_data.password, start_data.address,
-			*draw_control, texture_src, shader_src,
-			itemdef_manager, nodedef_manager, sound, eventmgr,
-			m_rendering_engine, connect_address.isIPv6(), m_game_ui.get());
+	try {
+		client = new Client(start_data.name.c_str(),
+				start_data.password, start_data.address,
+				*draw_control, texture_src, shader_src,
+				itemdef_manager, nodedef_manager, sound, eventmgr,
+				m_rendering_engine, connect_address.isIPv6(), m_game_ui.get());
+		client->migrateModStorage();
+	} catch (const BaseException &e) {
+		*error_message = fmtgettext("Error creating client: %s", e.what());
+		errorstream << *error_message << std::endl;
```

Comment:
`client` is still not freed if this code path is taken.

Reply:
I believe `client` is actually a member of `Game`, so it will be deleted in the destructor for `Game`.

Reply:
@TurkeyMcMac is correct, I can back it up from my own debug stepping